### DISCUSSION TITLE
how to use $eventcatcher with focus/blur on $edit-text widgets

### DISCUSSION THREAD

**Discussion by ericshulman** [TiddlyWiki/TiddlyWiki5#5912] [Q&A] [answered]
I am trying to use $event-catcher to trigger "side-effect" actions when an $edit-text widget gets or loses focus.  I tried the following test, but it doesn't seem to do anything:

```
\define showevent() <$action-log event=<<event-type>> />

<$eventcatcher events="focus blur" selector=".myclass" actions-focus=<<showevent>> actions-blur=<<showevent>>>
   <$edit-text class="myclass" tiddler="test" tag="input"/>
</$eventcatcher>
```

In contrast, if I use the following, it reports the mousedown event on the debugging console:
```
\define showevent() <$action-log event=<<event-type>> />
<$eventcatcher events="mousedown" selector=".myclass" actions-mousedown=<<showevent>>>
   <$edit-text class="myclass" tiddler="test" tag="input"/>
</$eventcatcher>
```

As an alternative, would it be reasonable to add "onFocus" and "onBlur" params to the $edit-text widget, so that I could write:
`<$edit-text class="myclass" tiddler="test" tag="input" onFocus=<<showevent>> onBlur=<<showevent>>/>`

**Comment by saqimtiaz** [comment, marked as answer]
Hi @ericshulman,

The underlying issue is that the focus and blur events don't bubble. I have used the `focusin` and `focusout` events in similar situations. Would this work for you for the time being?

```
\define showevent() <$action-log event=<<event-type>> />

<$eventcatcher events="focusin focusout" selector=".myclass" actions-focusin=<<showevent>> actions-focusout=<<showevent>>>
   <$edit-text class="myclass" tiddler="test" tag="input"/>
</$eventcatcher>
```
